browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -170,6 +186,9 @@ int OnBeforeURLRequest_SiteHacksWork(const ResponseCallback& next_callback,
   if (ctx->request_url.has_query()) {
     ApplyPotentialQueryStringFilter(ctx);
   }
+  if (ShouldBlockOnionAddress(ctx)) {
```

Comment:
This should be done in `OnionLocationNavigationThrottle::WillStartRequest` to prevent the request from being made
```
diff --git a/components/tor/onion_location_navigation_throttle.cc b/components/tor/onion_location_navigation_throttle.cc
index 482cea1bf2..cc030ca6da 100644
--- a/components/tor/onion_location_navigation_throttle.cc
+++ b/components/tor/onion_location_navigation_throttle.cc
@@ -97,10 +97,12 @@ OnionLocationNavigationThrottle::WillStartRequest() {
   // Open .onion site in Tor window
   if (!is_tor_profile_) {
     GURL url = navigation_handle()->GetURL();
-    if (url.SchemeIsHTTPOrHTTPS() && url.DomainIs("onion") &&
-        pref_service_->GetBoolean(prefs::kAutoOnionRedirect)) {
-      delegate_->OpenInTorWindow(navigation_handle()->GetWebContents(),
-                                 std::move(url));
+    if (url.SchemeIsHTTPOrHTTPS() && url.DomainIs("onion")) {
+      if (pref_service_->GetBoolean(prefs::kAutoOnionRedirect)) {
+        delegate_->OpenInTorWindow(navigation_handle()->GetWebContents(),
+                                   std::move(url));
+      }
+      return content::NavigationThrottle::BLOCK_REQUEST;
     }
   }
   return content::NavigationThrottle::PROCEED;
```